mysql-test/mysql-test-run.pl

```diff
@@ -968,6 +968,7 @@ ($$$)
           mtr_report("***Warnings generated in error logs during shutdown ".
                      "after running tests: $test_list\n\n$report");
           $test_failure= 1;
+          @$completed[-1]->{warnings}= $report;
```

Comment:
How did you test this patch?
I don't think you need this line, since if warning exist  test will now about.

Reply:
I specifically broke the test to test this. Test to check the result of this is the `galera_3nodes.galera_ipv6_mariabackup`.
If you rename the `bin/wsrep_sst_mariabackup` file and run the test, you will get a broken XML, like this:
```
<failure message="" type="MTR_RES_FAILED">
--
<![CDATA[ CURRENT_TEST: galera_ipv6_mariabackup ]]>
</failure>
```
Adding this line allows read these warnings from the other socket and output them in XML. Warnings and test results are generated at different stages: [results](https://github.com/MariaDB/server/blob/10.2/mysql-test/mysql-test-run.pl#L813), [warnings](https://github.com/MariaDB/server/blob/10.2/mysql-test/mysql-test-run.pl#L963).

Reply:
also tested in BB https://mdbe-buildbot.mariadb.net/#/builders/2/builds/3762 , see results https://server-test.mariadb.net/?products=MariaDBServerCommunity&show_skipped_results=false&only_last_commits=false&time_interval=-1&test_name=&commit_id=&ci_services=BuildBot&runs_with_additional_tools=true&excluded_target_build_ids=5832&excluded_target_build_ids=5817&excluded_target_build_ids=5838

Reply:
@xnox  Your change (in `mysql-test-run.pl`) has nothing to do with the test you tested.
Let me rephrase like this: 
what is a diff in case when you use that line (971) and when you don't use that line, in xml-report?
Note what is  a `$report` variable => `my $report= $fake_test->{'warnings'};`.
And where come from `$fake_test`? => `my $fake_test= My::Test::read_test($sock);` 
What means this? Is the test finished in this phase ? Are there any warnings?
When you say `other sockets` what do you mean, how is that connected to the single test case you showed (`galera`)?
@timofeyturenko can you please test patch [3757b9ffb5|https://github.com/MariaDB/server/commit/3757b9ffb5e0ff82704d2a30eab08cc7a20e9e53] and let me know the status?

Reply:
using https://github.com/MariaDB/server/commit/3757b9ffb5e0ff82704d2a30eab08cc7a20e9e53:

https://mdbe-ci-repo.mariadb.net/bb-logs/MariaDBServerCommunity/bb-10.2-anel-MDEV-24135-feb02-01/ubuntu/focal/mtr-normal-test/mtr.log
https://mdbe-ci-repo.mariadb.net/bb-logs/MariaDBServerCommunity/bb-10.2-anel-MDEV-24135-feb02-01/ubuntu/focal/mtr-normal-test/mtr.xml

XML does not contain warnings from Valgrind.